kataMarsRover.test.ts

```diff
@@ -0,0 +1,125 @@
+        [{width:5, length:5}, {x:3, y:0, head:'E' as direction}, "M", {"head": "E", "x": 4, "y": 0}],
+        [{width:5, length:5}, {x:4, y:0, head:'E' as direction}, "M", {"head": "E", "x": 5, "y": 0}],
+        [{width:5, length:5}, {x:5, y:0, head:'E' as direction}, "M", {"head": "E", "x": 5, "y": 0}]
+    ])(
+        'return final position and head of rover',
+        (initMap, initPos, drivePath, result) => {
+            expect(procMarsRover(initMap, initPos, drivePath)).toEqual(result)
+        }
+    )
+})
+
+describe('Validate output for move west RoverDriveModule', () => {
+    it.each([
+        [{width:5, length:5}, {x:5, y:0, head:'W' as direction}, "M", {"head": "W", "x": 4, "y": 0}],     
+        [{width:5, length:5}, {x:4, y:0, head:'W' as direction}, "M", {"head": "W", "x": 3, "y": 0}],     
+        [{width:5, length:5}, {x:3, y:0, head:'W' as direction}, "M", {"head": "W", "x": 2, "y": 0}],     
+        [{width:5, length:5}, {x:2, y:0, head:'W' as direction}, "M", {"head": "W", "x": 1, "y": 0}],
+        [{width:5, length:5}, {x:1, y:0, head:'W' as direction}, "M", {"head": "W", "x": 0, "y": 0}],
+        [{width:5, length:5}, {x:0, y:0, head:'W' as direction}, "M", {"head": "W", "x": 0, "y": 0}]
+    ])(
+        'return final position and head of rover',
+        (initMap, initPos, drivePath, result) => {
+            expect(procMarsRover(initMap, initPos, drivePath)).toEqual(result)
+        }
+    )
+})
+
+describe('Validate output for move south RoverDriveModule', () => {
+    it.each([
+        [{width:5, length:5}, {x:0, y:5, head:'S' as direction}, "M", {"head": "S", "x": 0, "y": 4}],     
+        [{width:5, length:5}, {x:0, y:4, head:'S' as direction}, "M", {"head": "S", "x": 0, "y": 3}],     
+        [{width:5, length:5}, {x:0, y:3, head:'S' as direction}, "M", {"head": "S", "x": 0, "y": 2}],     
+        [{width:5, length:5}, {x:0, y:2, head:'S' as direction}, "M", {"head": "S", "x": 0, "y": 1}],
+        [{width:5, length:5}, {x:0, y:1, head:'S' as direction}, "M", {"head": "S", "x": 0, "y": 0}],
+        [{width:5, length:5}, {x:0, y:0, head:'S' as direction}, "M", {"head": "S", "x": 0, "y": 0}]
+    ])(
+        'return final position and head of rover',
+        (initMap, initPos, drivePath, result) => {
+            expect(procMarsRover(initMap, initPos, drivePath)).toEqual(result)
+        }
+    )
+})
+
+describe('Validate output for MarsRoverProject', () => {
+    it.each([
+        [{width:5, length:5}, {x:1, y:2, head:'N' as direction}, "LMLMLMLMM", {"head": "N", "x": 1, "y": 3}],     
+        [{width:5, length:5}, {x:3, y:3, head:'E' as direction}, "MMRMMRMRRM", {"head": "E", "x": 5, "y": 1}]
+    ])(
+        'return final position and head of rover',
+        (initMap, initPos, drivePath, result) => {
+            expect(procMarsRover(initMap, initPos, drivePath)).toEqual(result)
+        }
+    )
+})
```

Comment:
I just have few questions on my mind regarding the tests:
1. What if a negative number is passed to  `roverMap`?
2. What if a negative number is passed to `roverLocator`?
3. What if the rover after moving is going to be out of the plateau range how the developer will know? is there like an error message or a message to show that? or you will return last valid data?

Let me know if you have thought about these questions and you would like to share your answers. If you need further information regarding my questions please let me know and I would love to help 🙂